src/content-helper/editor-sidebar/smart-linking/component.tsx

```diff
@@ -351,6 +352,28 @@ export const SmartLinkingPanel = ( {
 			return true;
 		} );
 
+		// Exclude Smart Links text and URLs that are already in the post, to avoid duplicates.
+		links = links.filter( ( link ) => {
+			return ! smartLinks.some( ( sl ) => {
+				if ( sl.href === link.href ) {
+					// eslint-disable-next-line no-console
+					console.warn( `PCH Smart Linking: Skipping duplicate link: ${ link.href }` );
+					return true;
+				}
+				if ( sl.text === link.text ) {
+					// If the offset is the same, we want to keep the link, so it can replace the old smart link.
+					if ( sl.offset === link.offset ) {
+						// TODO: Flag smart link as updated.
```

Comment:
Is this `TODO` something that needs to be implemented, or can it wait?

Reply:
It is something to be implemented in the future